src/system/components/handler.clj

```diff
@@ -27,20 +28,28 @@
  [endpoint]
   (reduce-kv (fn [_ k v] (if (contains+? v :middleware) (reduced k) _)) {} (val endpoint)))
 
-(defrecord Handler []
+(def ^:private accepted-routing-lib #{:compojure :bidi})
+
+(defmulti ^:private make-handler-fn accepted-routing-lib)
+
+(defmethod make-handler-fn :compojure [_] compojure/routes)
+
+(defmethod make-handler-fn :bidi [_] bidi/make-handler)
+
+(defrecord Handler [routing-lib]
   component/Lifecycle
   (start [component]
     (let [endpoints-with-middleware (partition-by middleware-key ((∘ with-middleware endpoints) component))
           handlers (for [endpoints endpoints-with-middleware
                          :let [mw-key (middleware-key (first endpoints))
                                wrap-mw (get-in (val (first endpoints)) [mw-key :wrap-mw])
                                routes (keep :routes (vals endpoints))]]
-                     (wrap-mw (apply compojure/routes routes)))
+                     (wrap-mw (apply (make-handler-fn routing-lib) routes)))
```

Comment:
Please rename `routing-lib` to `router`. It is a single Var that refers to the router library entry point. It can thus be used in a more straightforward way .